tests/providers/openai/hooks/test_openai.py

```diff
@@ -20,6 +20,9 @@
 from unittest.mock import patch
 
 import pytest
+
+openai = pytest.importorskip("openai")
```

Comment:
This and other related were needed because some of the providers also need Pydantic, and we remove those libraries from CI image that need it, so that these tests are skipped when we run tests in "no pydantic" or "pydantic 1" mode (pydantic 1 in case of pyiceberg because it also needs pydantic 2).